### PR TITLE
Add topology labels to endpoints

### PR DESCRIPTION
**Pull request by nmittler**
This implements approved RFC: https://docs.google.com/document/d/1ciVTulP-ClrJUcVKBKrbK_WHVjCfCXlymzjKP3sQW3A/edit

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go [hunk 1]
```diff
@@ -39,43 +39,69 @@ type EndpointBuilder struct {
...
 		sa = kube.SecureNamingSAN(pod)
 		uid = createUID(pod.Name, pod.Namespace)
-		podLabels = pod.Labels
+		endpointLabels = pod.Labels
 	}
 
 	return &EndpointBuilder{
 		controller:     c,
-		labels:         podLabels,
+		labels:         augmentLabels(endpointLabels, c.Cluster(), locality),
```
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go:54]: Why change it

File: pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go [hunk 2]
```diff
@@ -39,43 +39,69 @@ type EndpointBuilder struct {
...
 }
 
-func NewEndpointBuilderFromMetadata(c *Controller, proxy *model.Proxy) *EndpointBuilder {
+func NewEndpointBuilderFromMetadata(c controllerInterface, proxy *model.Proxy) *EndpointBuilder {
+	locality := util.LocalityToString(proxy.Locality)
 	return &EndpointBuilder{
 		controller:     c,
 		metaNetwork:    proxy.Metadata.Network,
-		labels:         proxy.Metadata.Labels,
+		labels:         augmentLabels(proxy.Metadata.Labels, c.Cluster(), locality),
```
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go:70]: same here

File: pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go (hunk 1 above)
Comment (nmittler) [reply pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go:54]: The motivation for adding labels is described in the RFC in the description.

**Comment by stevenctl** [comment]
Can we add an integration test with the DestinationRule example from the doc?

File: pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go:54]: make sense we can support routing within  same topology

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**